hack/chains/config.sh

```diff
@@ -78,6 +78,33 @@ case "$1" in
 
     ;;
 
+  rekor-local )
+    REKOR_SERVER=$(kubectl get ingress -n rekor-server -o yaml|yq e '.items[.spec].spec.rules[.host].host')
+    $0 "transparency.url: \"$REKOR_SERVER\"" $2
+
+    ;;
+
+  rekor-default )
+    # if setting rekor-default we should ensure we don't have a transparency.url value
+    $0 remove-key 'transparency.url' $2
+
+    ;;
+
+  remove-key )
+    # Remove the given key from the data section
+    set -x
+    kubectl patch $CHAINS_CONFIG --type=json -p='[{"op": "remove", "path":"'"/data/$2"'"}]'
+    $0 restart-controller
```

Comment:
Just for my own understanding, why is an explicit restart needed in this case but an implicit restart is ok on the other modifications?

Reply:
Because in the `remove-key` scenario it's a json type patch which removes the given key. As such we need to call the restart-controller explicitly there. In the other scenarios, which are merge patch scenarios which only merge (with no removals)  we're generating a $PATCH and then applying a merge patch operation to it and restarting the controller.

Reply:
Oh I see. Thank you for clarifying it.